### DISCUSSION TITLE
Nuxtjs precache components for offline use

### DISCUSSION THREAD

**Discussion by khatriafaz** [nuxt/nuxt#9540] [Questions (Nuxt 2)]
I am looking for a way by which I can precache all nuxtjs components for offline use, currently nuxtjs requests components in the viewport and then caches them for further offline use.

I want to fully precache all components so it can be used immediately even if it was not in the viewport

**Comment by kissu** [comment]
By using `nuxt/pwa`, you can look into [Workbox](https://developers.google.com/web/tools/workbox), which Nuxt specific documentation can be [found here](https://pwa.nuxtjs.org/workbox#offline). Never used it myself, but I think that this one will work fine.

If you have a feedback, I'd be interested into this myself! 🙏🏼

**Reply by khatriafaz** [reply]
I have used nuxt pwa version 2.6 so I can use precahing all routes functionality, it works correctly and precaches the routes js files, but it does not pre cache the components used in the routes pages, so it requests the components in runtime.

I want to pre cache all the components so all the files are ready to use offline on first request.
